### PR TITLE
add usage client to clientset

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/5826. I want to try to add a new collector for the quotas and see if the usage client is what we need.

### CODE REVIEW

File: client/azure_client_set.go [hunk 1]
```diff
@@ -14,44 +12,6 @@ import (
 	"github.com/giantswarm/microerror"
 )
 
-type AzureClientSetConfig struct {
```
Comment (xh3b4sd) [inline client/azure_client_set.go]: I moved this to its own file.

File: client/azure_client_set.go [hunk 2]
```diff
@@ -66,6 +26,8 @@ type AzureClientSet struct {
 	InterfacesClient *network.InterfacesClient
 	//StorageAccountsClient manages blobs in storage containers.
 	StorageAccountsClient *storage.AccountsClient
+	// UsageClient is used to work with limits and quotas.
+	UsageClient *compute.UsageClient
```
Comment (xh3b4sd) [inline client/azure_client_set.go]: Added the usage client here.

File: client/azure_client_set.go [hunk 3]
```diff
@@ -82,20 +44,21 @@ type AzureClientSet struct {
 
 // NewAzureClientSet returns the Azure API clients.
 func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
-	if err := config.Validate(); err != nil {
```
Comment (xh3b4sd) [inline client/azure_client_set.go]: Bad kitty.

File: client/azure_client_set.go [hunk 4]
```diff
@@ -82,20 +44,21 @@ type AzureClientSet struct {
...
-		return nil, microerror.Maskf(invalidConfigError, "config.%s", err)
+	err := config.Validate()
```
Comment (xh3b4sd) [inline client/azure_client_set.go]: Good kitty.

File: client/azure_client_set.go [hunk 5]
```diff
@@ -82,20 +44,21 @@ type AzureClientSet struct {
...
 	// cloud. For empty config.EnvironmentName returns Azure public cloud.
 	env, err := parseAzureEnvironment(config.EnvironmentName)
 	if err != nil {
-		return nil, microerror.Maskf(err, "parsing Azure environment")
+		return nil, microerror.Mask(err)
 	}
 
 	servicePrincipalToken, err := newServicePrincipalToken(config, env)
 	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
```
Comment (xh3b4sd) [inline client/azure_client_set.go]: I find these super useless to be honest. It does not add value during debugging.

File: client/fakeclient/azure.go [hunk 6]
```diff
@@ -1,15 +0,0 @@
-package fakeclient
-
-import (
-	"github.com/giantswarm/azure-operator/client"
-)
-
-// TODO(PK) remove it as soon as we sort AzureClient and calico CC extention.
-func NewAzureConfig() client.AzureClientSetConfig {
```
Comment (xh3b4sd) [inline client/fakeclient/azure.go]: I dropped this as it doesn't appear to be used anywhere anymore.

File: client/azure_client_set.go (hunk 5 above)
Comment (tuommaki) [reply client/azure_client_set.go]: As long as the original error has enough context for debugging, I agree.

Comment (xh3b4sd) [reply client/azure_client_set.go]: The stack is the context. At least this is for me the most relevant thing. The error message almost never tells me anything useful. I look at the stack. I look at the code. I fix the problem.

**Pull request merged**